.github/workflows/images.yaml

```diff
@@ -0,0 +1,51 @@
+# NUC image creation and push to DockerHub
+name: NUC Image Push
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push events for the main branch
+  push:
+    branches: [main]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Build and push all images
+  build_and_push:
+    name: "Build and Push Docker Images"
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        platform: ["generic", "nuc8i7beh", "nuc7i7bnh"]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      # Setup docker buildx
+      - name: 🐳 Set up docker buildx 🐳
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      # Build the docker image
+      - name: 🐳 Build the docker image 🐳
+        uses: docker/build-push-action@v2.7.0
+        with:
+          pull: true
+          tags: "nubots/nubots:${{ matrix.platform }}"
+          file: docker/Dockerfile
+          context: docker
+          build-args: |
+            platform=${{ matrix.platform }}
```

Comment:
I think we need `BUILDKIT_INLINE_CACHE=1` to use the buildkit inline cache exporter, as the docker one is virtually nonexistent.